jerry-core/parser/js/js-parser-statm.c

```diff
@@ -571,6 +574,21 @@ parser_parse_var_statement (parser_context_t *context_p) /**< context */
         cbc_opcode_t opcode = CBC_ASSIGN_SET_IDENT;
 
 #if ENABLED (JERRY_ES2015)
+        uint16_t function_literal_index = parser_check_anonymous_function_declaration (context_p);
+
+        if (function_literal_index < PARSER_NAMED_FUNCTION)
+        {
+          if (function_literal_index == PARSER_ANONYMOUS_CLASS)
+          {
+            lexer_construct_literal_object (context_p, &name_loc, LEXER_STRING_LITERAL);
+            parser_emit_cbc_ext_literal_from_token (context_p, CBC_EXT_SET_FUNCTION_NAME);
+          }
+          else
+          {
+            parser_set_function_name (context_p, function_literal_index, index, 0);
+          }
+        }
```

Comment:
This looks identical to the one in the `parser_pattern_form_assignment` method. Would it be possible to create a function for it?

Reply:
Not exactly. At his point the function name for anonymous classes can be constructed from the saved literal location, but in `parser_pattern_form_assignment ` we use `lexer_construct_string_literal_from_ident`.